### DISCUSSION TITLE
How to use new Force Fields with MDPow?

### DISCUSSION THREAD

**Discussion by eacoba** [Becksteinlab/MDPOW#208] [Q&A] [answered]
Thank your for your previous answers @orbeckst and @iorga at issues section. This time i would like to go further talking about GROMOS and SwissParam ff.  

1. I asked how to use GROMOS54A7 ff with MDPow, I have read that it is not hard to do but requires tedious changes to a few paths in template scripts, what is important to take in account? which scripts are critical? and what happen if I just change the files of a topology already included, for example of OPLS and put the GROMOS54A7 files. @iorga answer was:
> This might be possible, but it requires some work. The force field parameters from the GROMACS xxxx.ff folders are important, and should be imported in MDPOW, and also the water models. Then the specific options for Gromos should be added to different scripts, choice of the right water model by default, etc. Please open a discussion for this topic in the discussions forum if you want to go further.

I had some doubts. In mdpow ff files are imported using only the script `forcefields.py`? so I only need to edit the directories in `forcefields.py`, or is something carried out automatically with GROMACS wrapper? and then I have to copy the original ff in GROMACS xxxx.ff? even `fep.py` file should be modified?

2. Sometimes there are force fields that contain different or new atom types in the `itp` file, i think this is the case for OPLS, likewise for SwissParam, which I would like to use with `mdp` parameters for CHARMM, I have doubts about how to introduce an octanol force field, Could be enough to replace `octanol.itp` in the charmmxxxx.ff with a parametrized SwissParam octanol `itp`? Could you please give me an advice about this action? 
3. Finally, I would like to ask for a recommendation, to confirm that the changes in the python scripts are correct. I think that it would be enough to check that the `topol.top` and `mdp` files have the changes that I expect when I modify the script, but I would like to know if I should consider other files or elements of the code that I have not mentioned, and that you consider very relevant. Thank you very much for this section that you have enabled for questions.

**Comment by orbeckst** [comment, marked as answer]
1. I would create a whole bunch of new template files in [mdpow/templates](https://github.com/Becksteinlab/MDPOW/tree/develop/mdpow/templates), following the existing naming scheme.
2. add the FF directory to [mdpow.top](https://github.com/Becksteinlab/MDPOW/tree/develop/mdpow/top); MDPOW changes the GMXFF or GMXDATA (can't remember) env var to point to this directory instead of the standard one.
3. add the solvent coordinate files to `top`, too — see examples; you might have `1oct_GROMOS54A7.gro`
4. add the solvent itp files inside the FF directory (GROMOS54A7.ff/1oct.itp)
5. Then create new data structures in [forcefields.py](https://github.com/Becksteinlab/MDPOW/blob/develop/mdpow/forcefields.py),e.g. `GROMOS54A7_SOLVENT_MODELS` and add to https://github.com/Becksteinlab/MDPOW/blob/0ab82e4d88969e6004db9701ebc4af85ba2b3f6b/mdpow/forcefields.py#L199-L203  — This is all cumbersome and hacky because the original design did not anticipate using multiple force fields. Sorry.
6. Hard-code the path to your force field, ions, and water models in https://github.com/Becksteinlab/MDPOW/blob/0ab82e4d88969e6004db9701ebc4af85ba2b3f6b/mdpow/forcefields.py#L260-L272

This should be enough to run through the Python API and you would specify `forcefield="GROMOS54A7"` (the key that you used in `forcefields.py`) when setting up equilibrium and FEP simulations.

If you want to run the MDPOW scripts then it *should* just work by setting in your `runinput.yml` file the key
```yaml
setup:
   forcefield: "GROMOS54A7"
```

**Reply by eacoba** [reply]
Thank you for your answer @orbeckst, i'm going to follow your steps.
